src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -58,15 +58,15 @@ public override string GetCurrentManagedThreadStackTrace (int skipFrames, bool f
 		{
 			if (!reference.IsValid)
 				return null;
-			var peeked      = JNIEnvInit.ValueManager?.PeekPeer (reference);
+			var peeked      = JniEnvironment.Runtime.ValueManager.PeekPeer (reference);
```

Comment:
Changes like this we could also group in the separate PR.

Reply:
I needed to make this change because otherwise unit tests were failing. We have some tests which change the global `JniEnvironment.Runtime.ValueManager` instance to a different one and run tests against that value manager. During those tests, `JniEnvironment.Runtime.ValueManager` and `JNIEnvInit.ValueManager` would point to different value managers and things would start falling appart. I'm not sure this can easily be separated from the rest of the PR.

Reply:
I'm just thinking this change seems good on its own, and we could merge it early if we're waiting.

Reply:
I opened https://github.com/dotnet/android/pull/10222